src/ipc/ipc4/helper.c

```diff
@@ -235,15 +237,21 @@ int ipc_pipeline_free(struct ipc *ipc, uint32_t comp_id)
 static struct comp_buffer *ipc4_create_buffer(struct comp_dev *src, struct comp_dev *sink,
 					      uint32_t src_queue, uint32_t dst_queue)
 {
-	struct ipc4_base_module_cfg *src_cfg;
+	const struct comp_driver *drv = src->drv;
+	struct ipc4_base_module_cfg src_cfg;
 	struct comp_buffer *buffer = NULL;
 	struct sof_ipc_buffer ipc_buf;
-	int buf_size;
+	int buf_size, ret;
 
-	src_cfg = (struct ipc4_base_module_cfg *)comp_get_drvdata(src);
+	ret = drv->ops.get_large_config(src, IPC4_MODULE_BASE_CONFIG_PARAM_ID, true, true,
```

Comment:
need to add this for all supported ipc4 modules, like mixin | our,  SRC

Reply:
My suggestion would be to move this functionality to function get_attribute and return a pointer to ipc4_base_module_cfg. We will avoid unnecessary copying of the structure and if necessary, we will be able to cast the returned pointer to the type of configuration used by the given component type.

What do you think about it @lgirdwood @RanderWang @ranj063 @marcinszkudlinski @abonislawski @mwasko @mmaka1 ?

Reply:
@ranj063 https://github.com/thesofproject/sof/blob/main/src/include/sof/audio/component.h#L396

Reply:
@softwarecki I spoke too soon. get_attribute is OK too but like get_large_config, we cannot avoid the memcpy. Please see the last patch in the PR

Reply:
@ranj063  @lgirdwood  I think we are going in the right direction. These components also needs the get_attribute functionality: up_down_mixer, aria, src and asrc.
I don't know what about the `ipc4_comp_get_base_module_cfg` function. It should get components base configuration using the get_attribute mechanism but has no error handling. Maybe it should return an error code, or signal it by returning NULL that will be checked by caller?